src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2233,7 +2233,8 @@ private bool RefMustBeObeyed(bool isDelegateCreation, ArgumentSyntax argumentSyn
                 {
                     case BoundKind.PropertyAccess:
                     case BoundKind.IndexerAccess:
-                        hadError = !CheckValueKind(argumentSyntax, arg, BindValueKind.RefOrOut, false, diagnostics);
+                        var requiredValueKind = origRefKind == RefKind.In ? BindValueKind.ReadonlyRef : BindValueKind.RefOrOut;
```

Comment:
You'll want to use the other enum member, once you merge with Omar's change.

Reply:
done

---
In reply to: [141967622](https://github.com/dotnet/roslyn/pull/22405#discussion_r141967622) [](ancestors = 141967622)